web-client/src/modules/timeline/components/OffersList/OffersList.tsx

```diff
@@ -128,8 +132,9 @@ const OfferItem: React.FC<OfferItemProps> = ({
       <IconContainer>
         <EnvironmentOutlined />
       </IconContainer>
-      {/* TODO : Logic for using offer.address.coords for calculating the distance */}
-      <TextIcon>5 KM</TextIcon>
+      <TextIcon>
+        {haversineDistance(data.address.coords, requestCoords)}
```

Comment:
where do you get data from? did you intend for this to be `offer.address.coords`?